benches/hierarchy/mvcc/mod.rs

```diff
@@ -4,19 +4,23 @@ use criterion::{black_box, BatchSize, Bencher, Criterion};
 use kvproto::kvrpcpb::Context;
 use test_util::KvGenerator;
 use tikv::storage::kv::Engine;
-use tikv::storage::mvcc::{MvccReader, MvccTxn};
+use tikv::storage::mvcc::{MvccReader, MvccTxn, TimeStamp};
 use tikv::storage::{Key, Mutation, Options};
 
 use super::{BenchConfig, EngineFactory, DEFAULT_ITERATIONS, DEFAULT_KV_GENERATOR_SEED};
 
-fn setup_prewrite<E, F>(engine: &E, config: &BenchConfig<F>, start_ts: u64) -> (E::Snap, Vec<Key>)
+fn setup_prewrite<E, F>(
+    engine: &E,
+    config: &BenchConfig<F>,
+    start_ts: impl Into<TimeStamp>,
+) -> (E::Snap, Vec<Key>)
 where
     E: Engine,
     F: EngineFactory<E>,
 {
     let ctx = Context::default();
     let snapshot = engine.snapshot(&ctx).unwrap();
-    let mut txn = MvccTxn::new(snapshot, start_ts, true).unwrap();
+    let mut txn = MvccTxn::new(snapshot, start_ts.into(), true).unwrap();
```

Comment:
How about accepting `impl Into<TimeStamp>` as well in `MvccTxn::new()`, so that you don't need to write these `xxx.into()` ?

Reply:
I didn't want to do this in the general way because then we lost the benefit of type safety - the whole point of this is so that you can't do something like `MvccTxn::new(snapshot, region_id, true)`, which you could if we use `impl Into`.

Most of the annoying usage is in tests where we're using numeric literals, so I tried adding a test-only ctor, but because the tests are cross-crate, that doesn't work :-( I'm going to try using a macro, to restrict the conversion to literals, not sure if that will work well...

Reply:
It sort of works. What do you think?